opensearch-observability/src/main/kotlin/org/opensearch/observability/action/CreateCollaborationObjectRequest.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.observability.action
+
+import org.opensearch.action.ActionRequest
+import org.opensearch.action.ActionRequestValidationException
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.io.stream.Writeable
+import org.opensearch.common.xcontent.ToXContent
+import org.opensearch.common.xcontent.ToXContentObject
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.common.xcontent.XContentParser
+import org.opensearch.common.xcontent.XContentParserUtils
+import org.opensearch.commons.utils.fieldIfNotNull
+import org.opensearch.commons.utils.logger
+import org.opensearch.observability.model.BaseObjectData
+import org.opensearch.observability.model.CollaborationObjectDataProperties.createObjectData
+import org.opensearch.observability.model.CollaborationObjectDataProperties.getReaderForObjectType
+import org.opensearch.observability.model.CollaborationObjectType
+import org.opensearch.observability.model.RestTag.COLLABORATION_ID_FIELD
+import java.io.IOException
+
+/**
+ * Action request for creating new configuration.
+ */
+internal class CreateCollaborationObjectRequest : ActionRequest, ToXContentObject {
+    val collaborationId: String?
+    val type: CollaborationObjectType
+    val objectData: BaseObjectData?
+
+    companion object {
+        private val log by logger(CreateCollaborationObjectRequest::class.java)
+
+        /**
+         * reader to create instance of class from writable.
+         */
+        val reader = Writeable.Reader { CreateCollaborationObjectRequest(it) }
+
+        /**
+         * Creator used in REST communication.
+         * @param parser XContentParser to deserialize data from.
+         * @param id optional id to use if missed in XContent
+         */
+        @JvmStatic
+        @Throws(IOException::class)
+        fun parse(parser: XContentParser, id: String? = null): CreateCollaborationObjectRequest {
+            var collaborationId: String? = id
```

Comment:
use `val`? not sure if you need a separate variable if it's just `id`

Reply:
Removed redundant _collaborationId_ variable